dp/jobqueueing/gcsub.py

```diff
@@ -0,0 +1,164 @@
+from dp.jobqueueing.jobqueueing_db import GenerateClimosQueueEntry
+
+
+logger = default_logger()
+
+
+def make_qsub_test_script(queue_entry):
+    return '''
+#PBS -l nodes=1:ppn={qe.ppn}
+#PBS -l vmem={vmem}mb
+#PBS -l walltime={qe.walltime}
+#PBS -o {qe.output_directory}/logs
+#PBS -e {qe.output_directory}/logs
+#PBS -m abe
+#PBS -N generate_climos:{input_filename}
+#PBS -d /storage/home/rglover/code/climate-explorer-backend
+
+pbs_job_num=$(expr match "$PBS_JOBID" '\([0-9]*\)')
+
+# Set up the execution environment
+module load netcdf-bin
+module load cdo-bin
+source py3.4/bin/activate
+
+# Copy NetCDF file to $TMPDIR/climo/input
+indir=$TMPDIR/climo/input
+echo mkdir -p $indir && cp {qe.input_filepath} $indir
+infile=$indir/{input_filename}
+
+# Output directory is automatically created by generate_climos
+baseoutdir=$TMPDIR/climo/output
+outdir=$baseoutdir/$pbs_job_num
+
+# Generate climo means
+echo python dp/generate_climos.py -g {qe.convert_longitude} -v {qe.split_vars} -i {qe.split_intervals} -o $outdir $infile
+
+# Copy result file to final destination and remove temporary input file
+# Since output files are small, we're not removing them here.
+echo rsync -r $baseoutdir {qe.output_directory}
+echo rm $infile
+            '''.format(qe=queue_entry,
+                       input_filename=os.path.basename(queue_entry.input_filepath),
+                       vmem=queue_entry.ppn * 12000)
+
+
+def make_qsub_script(queue_entry):
+    return '''
+#PBS -l nodes=1:ppn={qe.ppn}
+#PBS -l vmem={vmem}mb
+#PBS -l walltime={qe.walltime}
+#PBS -o {qe.output_directory}/logs
+#PBS -e {qe.output_directory}/logs
+#PBS -m abe
+#PBS -N generate_climos:{input_filename}
+#PBS -d /storage/home/rglover/code/climate-explorer-backend
+set -o verbose
+
+pbs_job_num=$(expr match "$PBS_JOBID" '\([0-9]*\)')
+echo pbs_job_num = $pbs_job_num
+
+# Set up the execution environment
+module load netcdf-bin
+module load cdo-bin
+module list
+source py3.4/bin/activate
+which python
+
+# Copy NetCDF file to $TMPDIR/climo/input
+indir=$TMPDIR/climo/input
+echo indir = $indir
+mkdir -p $indir && cp {qe.input_filepath} $indir
+ls $indir
+infile=$indir/{input_filename}
+echo infile = $infile
+
+# Output directory is automatically created by generate_climos
+baseoutdir=$TMPDIR/climo/output
+echo baseoutdir = $baseoutdir
+outdir=$baseoutdir/$pbs_job_num
+echo outdir = $outdir
+
+# Generate climo means
+python dp/generate_climos.py -g {qe.convert_longitude} -v {qe.split_vars} -i {qe.split_intervals} -o $outdir $infile
+ls $outdir
+
+# Copy result file to final destination and remove temporary input file
+# Since output files are small, we're not removing them here.
+rsync -r $baseoutdir {qe.output_directory}
+ls $baseoutdir
+rm $infile
+            '''.format(qe=queue_entry,
+                       input_filename=os.path.basename(queue_entry.input_filepath),
+                       vmem=queue_entry.ppn * 12000)
+
+
+def submit_generate_climos_pbs_jobs(session, args):
```

Comment:
You might consider splitting `args` out into its child components that are actually used in the function (`number` and `test_job`, I think).

Absent doing so, there's nearly nothing in this file to indicate what the real input to this function is (without tracing the function and reading each place that `args` is used). Since the options to the `ArgumentParser` are hidden away in the helpers module.